--- a/solc/CommandLineInterface.cpp	2020-11-18 15:18:45.000000000 +0100
+++ b/solc/CommandLineInterface.cpp	2025-03-08 19:24:26.031436314 +0100
@@ -764,7 +764,8 @@
 
 void CommandLineInterface::createJson(string const& _fileName, string const& _json)
 {
-	createFile(boost::filesystem::basename(_fileName) + string(".json"), _json);
+	boost::filesystem::path _path (_fileName);
+	createFile(_path.stem().string() + string(".json"), _json);
 }
 
 bool CommandLineInterface::parseArguments(int _argc, char** _argv)
