--- a/liblangutil/SourceLocation.h	2020-11-18 14:18:45.000000000 +0000
+++ b/liblangutil/SourceLocation.h	2025-03-08 21:55:37.169028832 +0000
@@ -28,6 +28,7 @@
 
 #include <liblangutil/CharStream.h>
 
+#include <limits>
 #include <memory>
 #include <string>
 
@@ -117,7 +118,7 @@
 SourceLocation const parseSourceLocation(
 	std::string const& _input,
 	std::string const& _sourceName,
-	size_t _maxIndex = std::numeric_limits<size_t>::max()
+	size_t _maxIndex = std::max()
 );
 
 /// Stream output for Location (used e.g. in boost exceptions).
